rules/opentitan_test.bzl

```diff
@@ -270,6 +270,7 @@ def opentitan_functest(
         targets = VALID_TARGETS,
         args = [],
         data = [],
+        defines = [],
```

Comment:
The problem is that `defines` flow _up_ the dependency hierarchy (from leafs up to roots) rather than down.  In the context of compiling a program or test, you can think of the binary rule as a root of a subtree.

There are few ways you could approach this:
1) Pass a global flag like `--copt=...` when you want to build a debug version:
```
bazel build --copt=-DOTCRYPTO_STATUS_DEBUG //my/cryptolib:target
```

2) Create a config setting and add a `define` to the cc_library rule for cryptolib's `status.h`:
In `//sw/device/lib/crypto/impl/BUILD`:
```
config_setting(
    name = "cryptolib_debug",
    define_values = {
        "cryptolib_debug": "true"
    },
)

cc_library(
    name = "status",
    defines = select({
        ":cryptolib_debug": ["OTCRYPTO_STATUS_DEBUG"],
        "//conditions:default": [],
    }),
    ... the rest of the params ...
)
```
And then when you build:
```
bazel build //my/crypto:target --define cryptolib_debug=true
```

The second option has the advantage of not affecting `copt` across the whole codebase.  It should only affect cryptlib's `status` and dependent rules.

Reply:
Thank you! This is very helpful. I'll try one of the command-line options instead :slightly_smiling_face: